test/e2e/xfn_test.go

```diff
@@ -0,0 +1,318 @@
+/*
+Copyright 2023 The Crossplane Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package e2e
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"math/big"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/vladimirvivien/gexe"
+	"golang.org/x/exp/slices"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	client2 "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+	"sigs.k8s.io/e2e-framework/third_party/helm"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+
+	v1 "github.com/crossplane/crossplane/apis/apiextensions/v1"
+	"github.com/crossplane/crossplane/test/e2e/funcs"
+)
+
+func TestFunctions(t *testing.T) {
+
+	manifests := "test/e2e/manifests/apiextensions/composition/functions/private-registry"
+	environment.Test(t,
+		features.New("PullImageFromPrivateRegistryWithCustomCert").
```

Comment:
```suggestion
func TestFunctionsFromPrivateRegistryWithCustomCert(t *testing.T) {

	manifests := "test/e2e/manifests/apiextensions/composition/functions/private-registry"
	environment.Test(t,
		features.New("FunctionsFromPrivateRegistryWithCustomCert").
```

Reply:
Yeah, the test name function is to generic - I have another proposal - we are not testing here functions at all, rather xfn runner. Hence, I would like to rename `TestFunction` to `TestXfnRunner`.

Reply:
I would be more specific given that we are going for a 1:1 relationship between features and Test functions, we could say the Test function name should be the same as the feature name.

Reply:
how about that we then call the function as `TestXfnRunnerImagePull`?